crates/anstyle-svg/src/lib.rs

```diff
@@ -247,73 +245,100 @@ impl Term {
         )
         .unwrap();
         writeln!(&mut buffer, r#"  <style>"#).unwrap();
-        writeln!(&mut buffer, r#"    .{FG} {{ color: {fg_color} }}"#).unwrap();
-        writeln!(&mut buffer, r#"    .{BG} {{ background: {bg_color} }}"#).unwrap();
-        for (name, rgb) in color_styles(&elements, self.palette) {
+        self.render_classes(&mut buffer, &elements);
+        writeln!(&mut buffer, r#"  </style>"#).unwrap();
+        writeln!(&mut buffer, r#"</head>"#).unwrap();
+        writeln!(&mut buffer).unwrap();
+
+        if !self.background {
+            writeln!(&mut buffer, r#"<body>"#).unwrap();
+        } else {
+            writeln!(&mut buffer, r#"<body class="{BG}">"#).unwrap();
+        }
+        writeln!(&mut buffer).unwrap();
+        self.render_content(&mut buffer, styled_lines);
+        writeln!(&mut buffer).unwrap();
+
+        writeln!(&mut buffer, r#"</body>"#).unwrap();
+        writeln!(&mut buffer, r#"</html>"#).unwrap();
+        buffer
+    }
+
+    /// Returns the various parts needed to create an HTML page.
+    pub fn render_html_parts(&self, ansi: &str) -> HtmlParts {
```

Comment:
Is there a standard term for this?  I feel like I see the term "fragments" used.

Reply:
"fragments" sounds good too, gonna update the name then.